pkgs/by-name/sa/satdump/package.nix

```
@@ -0,0 +1,111 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  nix-update-script,
+  cmake,
+  pkg-config,
+  # required dependencies
+  fftwFloat,
+  libpng,
+  libtiff,
+  jemalloc,
+  volk,
+  nng,
+  curl,
+  # Optional dependencies
+  withZIQRecordingCompression ? true,
+  zstd,
+  withGUI ? true,
+  glfw,
+  zenity,
+  withAudio ? true,
+  portaudio,
+  withOfficialProductSupport ? true,
+  hdf5,
+  withOpenCL ? true,
+  opencl-headers,
+  ocl-icd,
+  withSourceRtlsdr ? true,
+  rtl-sdr-librtlsdr,
+  withSourceHackRF ? true,
+  hackrf,
+  withSourceAirspy ? true,
+  airspy,
+  withSourceAirspyHF ? true,
+  airspyhf,
+  withSourceAD9361 ? true,
+  libad9361,
+  libiio,
+  withSourceBladeRF ? true,
+  libbladeRF,
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "satdump";
+  version = "1.2.2";
+
+  src = fetchFromGitHub {
+    owner = "SatDump";
+    repo = "SatDump";
+    tag = finalAttrs.version;
+    hash = "sha256-+Sne+NMwnIAs3ff64fBHAIE4/iDExIC64sXtO0LJwI0=";
+  };
+
+  nativeBuildInputs = [
+    cmake
+    pkg-config
+  ];
+
+  buildInputs =
+    [
+      fftwFloat
+      libpng
+      libtiff
+      jemalloc
+      volk
+      nng
+      curl
+    ]
+    ++ lib.optional withZIQRecordingCompression zstd
+    ++ lib.optionals withGUI [
+      glfw
+      zenity
+    ]
+    ++ lib.optional withAudio portaudio
+    ++ lib.optional withOfficialProductSupport hdf5
+    ++ lib.optionals withOpenCL [
+      opencl-headers
+      ocl-icd
+    ]
+    ++ lib.optional withSourceRtlsdr rtl-sdr-librtlsdr
+    ++ lib.optional withSourceHackRF hackrf
+    ++ lib.optional withSourceAirspy airspy
+    ++ lib.optional withSourceAirspyHF airspyhf
+    ++ lib.optionals withSourceAD9361 [
+      libad9361
+      libiio
+    ]
+    ++ lib.optional withSourceBladeRF libbladeRF;
+
+  cmakeFlags = lib.optional (!withGUI) "-DBUILD_GUI=OFF";
```

Another nitpick (not a requirement to fix) I managed to miss on my `N` other reviews (sorry):

```suggestion
  cmakeFlags = [ (lib.cmakeBool "BUILD_GUI" withGUI) ];
```

Also, I'm a bit surprised that this isn't automatically detected based on the presence of `glfw`/`zenity`. While this is also **definitely** not required, it would be really cool if you could send a patch their way to add this auto-detection behavior.